### PR TITLE
Run jscodeshift programmatically

### PR DESCRIPTION
**Pull request by callingmedic911**
All credit goes to @dac09. Fixes `jscodeshift` issue: 
- On Node 16.
- Missing `jscodeshift` because it now resides in `root/node_modules/.bin` since Yarn 3.

@dac09 Now that `jscodeshift.run` returns promise, we need to handle it all the way - with async await. Any suggestion to minimize changes?

Test on:
- [ ] npx v7
- [ ] npx v8

### CODE REVIEW

**Comment by jtoar** [comment]
@callingmedic911 So happy to clean this up! But I have to ask only because this has caused us so much pain: have you tested it on macos/windows? And with npx v7 and v8?

**Comment by callingmedic911** [comment]
Using this programmatically should altogether bypass issues with different os and npx. I'll test to just make sure - on macos with npx v7 and v8. I'll need your help to get this tested on windows though. :)

Will keep you posted.

**Comment by jtoar** [comment]
@callingmedic911 Good point! And happy to help you test on Windows

**Comment by jtoar** [comment]
@callingmedic911 is there any reason to have the control flow in `runTransform` anymore? As far as I can tell the call to `jscodeshift.run` is the same in both blocks:

```js
export const runTransform = async ({
  transformPath,
  targetPaths,
  parser = 'tsx',
  options = {},
}: RunTransform) => {
  if (process.env.NODE_ENV === 'test') {
    try {
      await jscodeshift.run(transformPath, targetPaths, {
        ...defaultJscodeshiftOpts,
        parser,
        babel: process.env.NODE_ENV === 'test',
        ignorePattern: '**/node_modules/**',
        ...options, // Putting options here lets them override all the defaults.
      })
    } catch (e: any) {
      console.error('Transform Error', e.message)

      throw new Error('Failed to invoke transform')
    }
  } else {
    try {
      await jscodeshift.run(transformPath, targetPaths, {
        ...defaultJscodeshiftOpts,
        parser,
        babel: process.env.NODE_ENV === 'test',
        ignorePattern: '**/node_modules/**',
        ...options, // Putting options here lets them override all the defaults.
      })
    } catch (e: any) {
      console.error('Transform Error', e.message)

      throw new Error('Failed to invoke transform')
    }
  }
}
```

**Comment by jtoar** [comment]
Also so far this is working for me on Windows 🚀

**Comment by callingmedic911** [comment]
> is there any reason to have the control flow in runTransform anymore? As far as I can tell the call to jscodeshift.run is the same in both blocks:

I was too fixated on swapping things out I didn't realize it doesn't need control flow anymore .😅

**Comment by dac09** [comment]
LGTM! Just have to test this now

**Comment by dac09** [comment]
Pushed one other change @callingmedic911 - thanks for looking into this! 

I've verified using the RWJS debug method, on both node 16 and 14. I also made one small change to the deps (related to my babel changes earlier).

I now want to test it using npx - merging so I can run on canary.

File: packages/codemods/src/lib/runTransform.ts [hunk 1]
```diff
@@ -1,38 +1,28 @@
...
-      command: 'node',
-      cmdArgs: [jscodeshiftExecutable],
-    }
-  }
+const defaultJscodeshiftOpts = {
+  verbose: 0,
+  dry: false,
+  print: false,
+  babel: true,
+  extensions: 'js',
```
Comment (dac09) [inline packages/codemods/src/lib/runTransform.ts:17]: This extensions thing seems off, but according to the tests these seem to work.

File: packages/codemods/src/lib/runTransform.ts [hunk 2]
```diff
@@ -1,38 +1,28 @@
...
+  ignorePattern: '',
+  ignoreConfig: [],
+  runInBand: false,
+  silent: false,
+  parser: 'babel',
+  parserConfig: {},
+  failOnError: false,
```
Comment (dac09) [inline packages/codemods/src/lib/runTransform.ts:24]: I think perhaps we should set failOnError to be true. Not sure how to reproduce though to test

**Pull request merged**